### PR TITLE
[Doc] Mention how to work with Twig macros in Twig Components

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| Issues        | -
| License       | MIT

Made during the #SymfonyHackday

### CODE REVIEW

File: src/TwigComponent/doc/index.rst [hunk 1]
```diff
@@ -365,6 +365,19 @@ You can even give the block default content. See
...
+the full template path:
+
+    <twig:Alert>
+        {# ❌ this won't work #}
+        {% from _self import message_formatter %}
+        {# ✅ this works as expected #}
+        {% from 'some/path/template.html.twig' import message_formatter %}
+
+        {{ message_formatter('...') }}
+    </twig:Alert>
```
Comment (Kocal) [inline src/TwigComponent/doc/index.rst]: Not a RST expert here, but don't you need to use `.. code-block:: html+twig` to correctly highlight this code block?

Comment (javiereguiluz) [reply src/TwigComponent/doc/index.rst]: Hugo, you are absolutely right. I just fixed it. Thanks.

**Comment by Kocal** [comment]
Thanks Javier! :)

**Comment by smnandre** [comment]
@javiereguiluz @Kocal 

I feel this is [how macros work](https://twig.symfony.com/doc/3.x/tags/macro.html#macros-scoping) and not something related to HTML syntax (nor components in general)

I have tried quickly and none of this work: 

```twig

{# Classic embed #}
{% embed 'FooBar.html.twig' %}
    {% block content %}
        {% from _self import message_formatter as embed_formatter %}
        {{ embed_formatter('.abc.') }}
    {% endblock %}
{% endembed %}
    
{# Component template manually embeded #}    
{% embed 'components/Alert.html.twig' %}
    {% block content %}
         ❌ Will not work..  
        {% from _self import message_formatter as embed_formatter %}
        {{ embed_formatter('.abc.') }}
    {% endblock %}
{% endembed %}

{# Component tag with implicit "content" block #}        
{% component 'Alert' %}
      ❌ Will not work..  
    {% from _self import message_formatter as component_formatter %}
    {{ component_formatter('.abc.') }}
{% endcomponent %}

{# Component tag with explicit "content" block #}            
{% component 'Alert' %}
    {% block content %}
          ❌ Will not work..  
        {% from _self import message_formatter as component_formatter %}
        {{ component_formatter('.abc.') }}
    {% endblock %}
{% endcomponent %}
```

Did something in particular made you guys think it was due to HTML syntax? 

Did i miss something here ?

**Comment by Kocal** [comment]
> Did something in particular made you guys think it was due to HTML syntax?

I don't think so.

@javiereguiluz do you have more info about that?

**Comment by javiereguiluz** [comment]
OK, sorry for the confusion 😐  Should I create a new PR to revert these changes or can you revert it yourself? Thanks.

**Comment by Kocal** [comment]
No worries, I should have been more careful too!

Can you revert the PR? Usually I use the button "Revert" from GitHub UI, but I do not see it :/

**Comment by javiereguiluz** [comment]
Changes reverted in #2437.

**Comment by smnandre** [comment]
No need to be sorry of anything 😃  this kind of different behaviours between component tags and HTML syntax is quite frequent.. I was more afraid that something had changed recently / worked before "by chance" etc. 

❤️

**Pull request merged**